packages/client/src/containers/MovieDetails/MovieDetails.jsx

```diff
@@ -1,19 +1,33 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './MovieDetails.Style.css';
 import { Container } from '@mui/material';
 import BigMovieCard from '../../components/BigMovieCard/BigMovieCard';
 import TopCastDisplay from '../../components/TopCastDisplay/TopCastDisplay';
 import SimilarMoviesCarousel from '../../components/SimilarMoviesCarousel/SimilarMoviesCarousel';
 import { useParams } from 'react-router-dom';
+import { apiURL } from '../../apiURL';
 
 export const MovieDetails = () => {
   const movieID = useParams().id;
-
+  const [currentMovie, setCurrentMovie] = useState(undefined);
+  useEffect(() => {
+    (async () => {
+      const response = await fetch(`${apiURL()}/movies/${movieID}/details`);
+      const movie = await response.json();
+      if (movie) {
+        setCurrentMovie(movie);
+      }
+    })();
+  }, [movieID]);
```

Comment:
added a guard here as well, but I suspect that here it is not so important, because movieID comes from URL and the component is only displayed when the route is /movies/:id . So if there is no id in URL this component will not render and the app will fall back to other page for the /* route or /movies route

Reply:
Oh! I've just tested the route by literally typing in "movies/:id" in the address bar - and without a guard suggested by you it failed miserably :)
I have to take back my previous comment. Thank you once again for your remarks!